<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>IrisCC</class>
 <widget class="QMainWindow" name="IrisCC">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1077</width>
    <height>829</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>9999999</width>
    <height>999999</height>
   </size>
  </property>
  <property name="focusPolicy">
   <enum>Qt::ClickFocus</enum>
  </property>
  <property name="windowTitle">
   <string>Iris Camera Calibration</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../res/IrisCC.qrc">
    <normaloff>:/icon/lib-iris.png</normaloff>:/icon/lib-iris.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_9">
    <property name="leftMargin">
     <number>7</number>
    </property>
    <item>
     <widget class="QTabWidget" name="scene">
      <property name="tabPosition">
       <enum>QTabWidget::South</enum>
      </property>
      <property name="tabShape">
       <enum>QTabWidget::Rounded</enum>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab_image">
       <attribute name="title">
        <string>Image</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_7">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="margin">
         <number>0</number>
        </property>
        <item>
         <widget class="QCustomPlot" name="plot_image" native="true"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_scene">
       <attribute name="title">
        <string>Scene</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_7">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="margin">
         <number>0</number>
        </property>
        <item>
         <widget class="WorldPoses" name="plot_poses" native="true"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_cameras">
       <attribute name="title">
        <string>Cameras</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout">
        <property name="margin">
         <number>10</number>
        </property>
        <item row="1" column="0">
         <widget class="QGroupBox" name="groupBox_7">
          <property name="title">
           <string>Intrinsics:</string>
          </property>
          <layout class="QFormLayout" name="formLayout_2">
           <property name="fieldGrowthPolicy">
            <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
           </property>
           <item row="0" column="0">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Image Size:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLabel" name="image_size">
             <property name="minimumSize">
              <size>
               <width>100</width>
               <height>0</height>
              </size>
             </property>
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Focal Length:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLabel" name="focal_length">
             <property name="minimumSize">
              <size>
               <width>100</width>
               <height>0</height>
              </size>
             </property>
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_6">
             <property name="text">
              <string>Principal Point</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QLabel" name="principal_point">
             <property name="minimumSize">
              <size>
               <width>100</width>
               <height>0</height>
              </size>
             </property>
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item row="3" column="0" colspan="2">
            <widget class="QPlainTextEdit" name="intrinsic_matrix">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="backgroundVisible">
              <bool>false</bool>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QGroupBox" name="groupBox_6">
          <property name="title">
           <string>Camera</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_6">
           <item>
            <widget class="QComboBox" name="camera"/>
           </item>
          </layout>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QGroupBox" name="groupBox_9">
          <property name="title">
           <string>Extrinsics:</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_8">
           <item>
            <widget class="QPlainTextEdit" name="extrinsic_matrix">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QGroupBox" name="groupBox_10">
          <property name="title">
           <string>Distortion</string>
          </property>
          <layout class="QFormLayout" name="formLayout_3">
           <item row="0" column="0">
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Model:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>Params:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLineEdit" name="distortionParams"/>
           </item>
          </layout>
         </widget>
        </item>
        <item row="0" column="1" rowspan="4">
         <widget class="QGroupBox" name="groupBox_8">
          <property name="title">
           <string>Reprojection Error:</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <item>
            <widget class="QCustomPlot" name="plot_error" native="true">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>100</width>
               <height>100</height>
              </size>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QFormLayout" name="formLayout_4">
             <item row="0" column="0">
              <widget class="QLabel" name="label_8">
               <property name="text">
                <string>Error:</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLabel" name="error">
               <property name="text">
                <string/>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="label_9">
               <property name="text">
                <string>Variance:</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QLabel" name="variance">
               <property name="text">
                <string/>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QGroupBox" name="groupBox_5">
        <property name="title">
         <string>Input</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QPushButton" name="load">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Load Images</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <layout class="QFormLayout" name="formLayout">
             <item row="0" column="0">
              <widget class="QLabel" name="label">
               <property name="text">
                <string>Camrea ID:</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QSpinBox" name="cameraID">
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
               <property name="maximum">
                <number>999999999</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QToolButton" name="clear">
             <property name="autoFillBackground">
              <bool>false</bool>
             </property>
             <property name="styleSheet">
              <string notr="true"/>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="icon">
              <iconset resource="../res/IrisCC.qrc">
               <normaloff>:/icon/process-stop.svg</normaloff>:/icon/process-stop.svg</iconset>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_3">
        <property name="title">
         <string>Images</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QListWidget" name="image_list">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="title">
         <string>Finder</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_11">
         <item>
          <widget class="QComboBox" name="select_finder">
           <item>
            <property name="text">
             <string>None</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Chessboard</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Uchiyamama</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Random Features (Experimental)</string>
            </property>
           </item>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="configure_finder">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="icon">
            <iconset resource="../res/IrisCC.qrc">
             <normaloff>:/icon/applications-system.svg</normaloff>:/icon/applications-system.svg</iconset>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="title">
         <string>Calibration</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <item>
            <widget class="QComboBox" name="select_calibration">
             <property name="currentIndex">
              <number>0</number>
             </property>
             <item>
              <property name="text">
               <string>None</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>OpenCV Single Camera</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>OpenCV Stereo</string>
              </property>
             </item>
            </widget>
           </item>
           <item>
            <widget class="QToolButton" name="configure_calibration">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="icon">
              <iconset resource="../res/IrisCC.qrc">
               <normaloff>:/icon/applications-system.svg</normaloff>:/icon/applications-system.svg</iconset>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QPushButton" name="update">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="styleSheet">
            <string notr="true"/>
           </property>
           <property name="text">
            <string>Update</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_4">
        <property name="title">
         <string>Output</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QPushButton" name="save">
           <property name="text">
            <string>Save</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header location="global">qcustomplot.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>WorldPoses</class>
   <extends>QWidget</extends>
   <header location="global">WorldPoses.hpp</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../res/IrisCC.qrc"/>
 </resources>
 <connections/>
</ui>
